### PR TITLE
feat: allow user config AsyncLogger in log4j2

### PR DESCRIPTION
**Pull request by shoothzj**
### Motivation

Adding lmax disruptor in dependency allow user to config AsyncLogger in log4j2.xml like this

```xml
    <Loggers>
        <AsyncLogger name="async" level="info">
            <AppenderRef ref="AsyncConsole"/>
        </AsyncLogger>
    </Loggers>
```

See also https://logging.apache.org/log4j/2.x/manual/async.html

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -147,8 +150,6 @@
     <junit.version>4.13.2</junit.version>
     <hamcrest.version>1.3</hamcrest.version>
     <junit5.version>5.10.2</junit5.version>
-    <assertj-core.version>3.25.3</assertj-core.version>
-    <awaitility.version>4.2.0</awaitility.version>
```
Comment (lhotari) [inline pom.xml:151]: I'd suggest to not move these in the same PR. If there's a need to put these in alphabetical order, it's better to do that separately and do it consistently. Currently the testing related libraries are kept together and that's why assertj-core and awaitility versions are after junit5 version.

Comment (shoothzj) [reply pom.xml]: @lhotari Okay

**Review by lhotari — APPROVED** [review]
LGTM

**Comment by shoothzj** [comment]
@nicoloboschi @eolivelli @hangc0276 PTAL

**Comment by shoothzj** [comment]
@nicoloboschi @eolivelli @hangc0276 PTAL

**Review by jiazhai — APPROVED** [review]

**Pull request merged**